### PR TITLE
add secondary recovery date and user key management

### PR DESCRIPTION
**Pull request by calellowitz**
This implements the web portion of [DB Key](https://docs.google.com/document/d/1VCFGdJ-2JvfSerMsg9QvKF3kCraqt_UrdE1DBu7S-Zw/edit) and [Secondary Phone Verification](https://docs.google.com/document/d/1CjpB6mSTF-tXJsdJIMmfUwCHmB7nWYFhZXtv0UHGWYw/edit)

### CODE REVIEW

File: users/models.py [hunk 1]
```diff
@@ -36,6 +43,24 @@ def check_recovery_pin(self, pin):
...
 
+class UserKey(models.Model):
+    user = models.ForeignKey(ConnectUser, on_delete=models.CASCADE)
+    key = models.TextField()
+    valid = models.BooleanField(default=True)
+    created = models.DateTimeField(auto_now_add=True)
+
+    @classmethod
+    def get_or_create_key_for_user(cls, user):
+        now = now()
```
Comment (pxwxnvermx) [inline users/models.py:54]: Is this required? This is not getting used anywhere in the method.

File: users/models.py [hunk 2]
```diff
@@ -36,6 +43,24 @@ def check_recovery_pin(self, pin):
         return check_password(pin, self.recovery_pin)
 
 
+class UserKey(models.Model):
+    user = models.ForeignKey(ConnectUser, on_delete=models.CASCADE)
+    key = models.TextField()
```
Comment (pxwxnvermx) [inline users/models.py:48]: Can we assign a max_length for this field? I ran the logic from `get_or_create_key_for_user` locally and after the `b64encode` the key length comes out to be 44 characters.

File: users/models.py (hunk 1 above)
Comment (calellowitz) [reply users/models.py:54]: good catch

File: users/models.py (hunk 2 above)
Comment (calellowitz) [reply users/models.py:48]: yeah, thats a good idea. will pad a bit for safety but a limit seems good

**Review by pxwxnvermx — APPROVED** [review]

**Pull request merged**